Doc/c-api/unicode.rst

```diff
@@ -1035,6 +1035,15 @@ These are the UTF-8 codec APIs:
 
    As :c:func:`PyUnicode_AsUTF8AndSize`, but does not store the size.
 
+   .. warning::
+
+      This function does not have any special behavior for
+      `null bytes <https://en.wikipedia.org/wiki/Null_character>`_ embedded within
+      *unicode*. As a result, strings containing null bytes will remain in the returned
+      string, which some C functions might interpret as the end of the string, leading to
+      truncation. When handling user input, it is recommended to use :c:func:`PyUnicode_AsUTF8AndSize`
```

Comment:
I don't think that "user input" is useful here. I would prefer to say something like: "If truncation is an issue, ..."